src/tree/ResolvableTreeItemBase.ts

```diff
@@ -63,6 +63,13 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
         });
     }
 
+    public async pickTreeItemImpl(expectedContextValues: (string | RegExp)[], context: IActionContext): Promise<AzExtTreeItem | undefined> {
+        if (!this.resolveResult) {
+            await this.resolve(true, context);
+        }
+        return super.pickTreeItemImpl?.(expectedContextValues, context);
+    }
```

Comment:
Should we be resolving these in a `pickTreeItemImpl` in `GroupTreeItemBase`?

Reply:
Yeah, I think that makes more sense.

We can either have a loading bar while the children resolve or just display the list of current children and have them resolve in the background, and then await it if it hasn't finished resolving by the time the user selects it.